front/src/components/LanguageChanger/LanguageChanger.test.js

```diff
@@ -0,0 +1,44 @@
+import React from 'react';
+import { render } from '@testing-library/react';
+import { LanguageChanger } from './index';
+import { Logo } from '../Logo';
+import '../../i18n';
+import '@testing-library/jest-dom/extend-expect';
+
+describe('Language changer', () => {
+  test('should be default language', () => {
+    const { container } = render(
+      <LanguageChanger defaultLanguage={'en'} />
+    );
+    expect(container).toMatchSnapshot();
+  });
+  test('should be list of languages', () => {
+    const { container, getByText } = render(
+      <LanguageChanger defaultLanguage={'en'} />
+    );
+    const button = container.querySelector('.fa-chevron-down')
+      .parentElement;
+    button.click();
+    expect(getByText('English')).toBeInTheDocument();
+    expect(getByText('Русский')).toBeInTheDocument();
+    expect(getByText('Česky')).toBeInTheDocument();
+    expect(container).toMatchSnapshot();
+  });
+  test('should change language to ru', () => {
```

Comment:
Nice test. I'm slightly concerned about calling render for `<LanguageChanger/>` and `<Logo />` separately although they are the part of the same app. Can we somehow combine them? Can we even avoid bringing `<Logo />` as a dependency into that test?

Reply:
IMO this even looks more like integration test. What do you think about that?